server/src/test/java/io/crate/user/TransportRoleActionTest.java

```diff
@@ -0,0 +1,143 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.user;
+
+import static io.crate.testing.Asserts.assertThat;
+import static io.crate.user.metadata.RolesDefinitions.DUMMY_USERS;
+import static io.crate.user.metadata.RolesDefinitions.DUMMY_USERS_AND_ROLES;
+import static io.crate.user.metadata.RolesDefinitions.SINGLE_USER_ONLY;
+import static io.crate.user.metadata.RolesDefinitions.getSecureHash;
+import static io.crate.user.metadata.RolesDefinitions.usersMetadataOf;
+
+import java.util.Map;
+
+import org.elasticsearch.cluster.metadata.Metadata;
+import org.elasticsearch.test.ESTestCase;
+import org.junit.Test;
+
+import io.crate.user.metadata.RolesMetadata;
+import io.crate.user.metadata.UsersMetadata;
+import io.crate.user.metadata.UsersPrivilegesMetadata;
+
+public class TransportRoleActionTest extends ESTestCase {
+
+    @Test
+    public void testCreateFirstUser() throws Exception {
+        Metadata.Builder mdBuilder = new Metadata.Builder();
+        TransportCreateRoleAction.putRole(mdBuilder, "root", true, null);
+        RolesMetadata metadata = (RolesMetadata) mdBuilder.getCustom(RolesMetadata.TYPE);
+        assertThat(metadata.roleNames()).containsExactly("root");
+    }
+
+    @Test
+    public void testEmptyPrivilegesAreCreatedForNewUsers() throws Exception {
+        Metadata.Builder mdBuilder = new Metadata.Builder();
+        TransportCreateRoleAction.putRole(mdBuilder, "root", true, null);
+        UsersPrivilegesMetadata metadata = (UsersPrivilegesMetadata) mdBuilder.getCustom(UsersPrivilegesMetadata.TYPE);
+        assertThat(metadata.getUserPrivileges("root")).isEmpty();
+    }
+
+    @Test
+    public void testCreateUserAlreadyExists() throws Exception {
+        Metadata.Builder mdBuilder = new Metadata.Builder()
+            .putCustom(RolesMetadata.TYPE, new RolesMetadata(SINGLE_USER_ONLY));
+        assertThat(TransportCreateRoleAction.putRole(mdBuilder, "Arthur", true, null)).isTrue();
+    }
+
+    @Test
+    public void testCreateUser() throws Exception {
+        Metadata.Builder mdBuilder = new Metadata.Builder()
+            .putCustom(RolesMetadata.TYPE, new RolesMetadata(SINGLE_USER_ONLY));
+        TransportCreateRoleAction.putRole(mdBuilder, "Trillian", true, null);
+        RolesMetadata newMetadata = (RolesMetadata) mdBuilder.getCustom(RolesMetadata.TYPE);
+        assertThat(newMetadata.roleNames()).containsExactlyInAnyOrder("Trillian", "Arthur");
+    }
+
+    @Test
+    public void test_create_user_with_old_users_metadata() throws Exception {
```

Comment:
I should have separated the rename of the class and the new tests:
This is new.